Handling Events page translation (#37)
reactjs/hu.reactjs.org: content/docs/handling-events.md

---
id: handling-events
title: Események kezelése
permalink: docs/handling-events.html
prev: state-and-lifecycle.html
next: conditional-rendering.html
redirect_from:
  - "docs/events-ko-KR.html"
---

A React elemek eseményeinek kezelése nagyon hasonló a DOM elemek eseménykezeléséhez. Viszont van néhány szintaxisbeli különbség: 

* A React események elnevezésére camelCase-t használunk kisbetűk helyett.
* A JSX-ben string helyett függvényt adunk át az eseménykezelőnek.

Például ez a HTML:

```html
<button onclick="activateLasers()">
  Lézerek aktiválása
</button>
```

kissé máshogyan néz ki Reactben:

```js{1}
<button onClick={activateLasers}>
  Lézerek aktiválása
</button>
```

> Egy másik különbség, hogy nem tudsz `false` értéket visszaadni, hogy megakadályozd a default viselkedést Reactben. Expliciten meg kell hívni a `preventDefault`-ot. Például szimpla HTML-ben ahhoz, hogy megakadályozzuk egy linknek a default viselkedését, ezt írhatjuk:

Reviewer: ```suggestion
Egy másik különbség, hogy Reactben `false` érték visszaadásával nem tudod megakadályozni az alap viselkedést. Határozottan meg kell hívni a `preventDefault`-ot. Egyszerű HTML-ben például egy link alap viselkedésének megváltoztatásához írhatjuk ezt:
```

);
```

[**Próbáld ki CodePenen**](https://codepen.io/gaearon/pen/xEmzGg?editors=0010)

> Vigyázni kell a `this` használatával a JSX callbackekben. JavaScriptben az osztálymetódusok alapesetben nincsenek [hozzákötve](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind) a `this`-hez. Ha elfelejtenéd hozzákötni a `this.handleClick`-et a `this`-hez és így rendeled hozzá az `onClick`-hez, a `this` `undefined` értékű lesz amikor a metódus meghívódik.

Reviewer: ```suggestion
Légy óvatos a `this` használatával a JSX callbackekben. A JavaScriptben az osztálymetódusok alapesetben nincsenek [hozzákötve](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_objects/Function/bind) a `this`-hez. Ha elfelejtenéd hozzákötni a `this.handleClick`-et a `this`-hez és így rendeled hozzá az `onClick`-hez, a `this` `undefined` értékű lesz amikor a metódus meghívódik.
```

  }

  render() {
    // Ez a szintaxis biztosítja a `this` handleClickhez való kötését
    return (
      <button onClick={(e) => this.handleClick(e)}>
        Kattints rám
      </button>
    );
  }
}
```

> Az a probléma ezzel a szintaxissal, hogy a `LoggingButton` minden egyes renderelésekor egy új callback hozódik létre. A legtöbb esetben ez nem gond. Viszont ha ezt a callbacket gyerekkomponenseknek adjuk át, azok lehet, hogy még egyszer újrarenderelődnek. Mi általánosságban a konstruktor bindingot vagy az oszálymező szintaxist ajánljuk, hogy elkerüljük az efféle teljesítményproblémákat.

Reviewer: ```suggestion
A probléma ezzel a szintaxissal az, hogy a `LoggingButton` minden egyes renderelésekor egy új callbacket hozunk létre. A legtöbb esetben ez nem gond. Viszont ha ezt a callbacket gyermek komponenseknek adjuk tovább, azok feleslegesen renderelődhetnek újra. Az efféle teljesítményproblémák elkerülése érdekében általában a konstruktoron belüli kötést vagy az osztálymező szintaxis használatát ajánljuk.
```
